app/javascript/dashboard/routes/dashboard/settings/personal/NotificationPreferences.vue

```diff
@@ -0,0 +1,321 @@
+            :value="`${type}_${notification.value}`"
+            :span="type === 'push' ? 3 : 2"
+            :selected-flags="
+              type === 'email' ? selectedEmailFlags : selectedPushFlags
+            "
+            @input="handleInput"
+          />
+        </div>
+      </div>
+    </div>
+    <!--  Layout for mobile devices -->
+    <div class="flex flex-col gap-6 sm:hidden">
+      <span class="text-sm font-medium uppercase text-ash-900">
+        {{ $t('PROFILE_SETTINGS.FORM.EMAIL_NOTIFICATIONS_SECTION.TITLE') }}
+      </span>
+      <div class="flex flex-col gap-4">
+        <div
+          v-for="(notification, index) in filteredNotificationTypes"
+          :key="index"
+          class="flex flex-row items-start gap-2"
+        >
+          <notification-check-box
+            type="email"
+            :value="`email_${notification.value}`"
+            :selected-flags="selectedEmailFlags"
+            @input="handleEmailInput"
+          />
+          <span class="text-sm text-ash-900">{{ $t(notification.label) }}</span>
+        </div>
+      </div>
+
+      <div class="flex items-center justify-start gap-2">
+        <span class="text-sm font-medium uppercase text-ash-900">
+          {{ $t('PROFILE_SETTINGS.FORM.PUSH_NOTIFICATIONS_SECTION.TITLE') }}
+        </span>
+        <form-switch
+          :value="hasEnabledPushPermissions"
+          @input="onRequestPermissions"
+        />
+      </div>
+
+      <div class="flex flex-col gap-4">
+        <div
+          v-for="(notification, index) in filteredNotificationTypes"
+          :key="index"
+          class="flex flex-row items-start gap-2"
+        >
+          <notification-check-box
+            type="push"
+            :value="`push_${notification.value}`"
+            :selected-flags="selectedPushFlags"
+            @input="handleInput"
+          />
+          <span class="text-sm text-ash-900">{{ $t(notification.label) }}</span>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import alertMixin from 'shared/mixins/alertMixin';
+import configMixin from 'shared/mixins/configMixin';
+import uiSettingsMixin from 'dashboard/mixins/uiSettings';
+import TableHeaderCell from 'dashboard/components/widgets/TableHeaderCell.vue';
+import {
+  hasPushPermissions,
+  requestPushPermissions,
+  verifyServiceWorkerExistence,
+} from 'dashboard/helper/pushHelper.js';
+import { FEATURE_FLAGS } from 'dashboard/featureFlags';
+import FormSwitch from 'v3/components/Form/Switch.vue';
+import NotificationCheckBox from './NotificationCheckBox.vue';
+
+export default {
+  components: {
+    TableHeaderCell,
+    FormSwitch,
+    NotificationCheckBox,
+  },
+  mixins: [alertMixin, configMixin, uiSettingsMixin],
+  data() {
+    return {
+      selectedEmailFlags: [],
+      selectedPushFlags: [],
+      enableAudioAlerts: false,
+      hasEnabledPushPermissions: false,
+      notificationTypes: [
```

Comment:
See if we can move it to a constant file. So, the component size will be reduced.